### PR TITLE
SWP-1608 Improve CurrencyHelper facade

### PR DESCRIPTION
**Pull request by francescopepe**
- Add currencies default precision in a dedicated config file.
- Add function to CurrencyHelper facade to return the standard currency format, with the default currency precision.
- Add tests.

### CODE REVIEW

File: src/CurrencyHelper.php [hunk 1]
```diff
@@ -2,29 +2,39 @@
...
+     */
+    const MAX_PRECISION = 6;
+
     /**
      * Return a value in the given currency formatted for the given locale.
      *
      * @param float|int $value
      * @param string $currencyCode
      * @param string $locale
      *
```
Comment (AndreiaMatiasSilva) [inline src/CurrencyHelper.php:22]: no need for this empty line

File: src/CurrencyHelper.php [hunk 2]
```diff
@@ -2,29 +2,39 @@
...
+     * @param int $precision
      * @return string
```
Comment (AndreiaMatiasSilva) [inline src/CurrencyHelper.php:24]: we usually leave an empty line before @return

File: src/CurrencyHelper.php [hunk 3]
```diff
@@ -89,4 +105,18 @@ public function getSymbol(string $currencyCode = CurrencyCode::POUND_STERLING, s
 
         return $formatter->getSymbol(NumberFormatter::CURRENCY_SYMBOL);
     }
+
+    public function toStandardFormat(
```
Comment (AndreiaMatiasSilva) [inline src/CurrencyHelper.php:109]: missing docblock

File: src/Facades/CurrencyHelper.php [hunk 4]
```diff
@@ -6,10 +6,13 @@
 use Illuminate\Support\Facades\Facade;
 
 /**
+ * // phpcs:disable
```
Comment (AndreiaMatiasSilva) [inline src/Facades/CurrencyHelper.php:9]: why do we need to disable `phpcs` here?

File: src/config/currencies.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+return [
+    CurrencyCode::POUND_STERLING => [
+        'precision' => 2,
+    ],
+    CurrencyCode::EURO => [
+        'precision' => 2,
+    ],
+    CurrencyCode::NORWEGIAN_KRONE => [
+        'precision' => 2,
+    ],
```
Comment (AndreiaMatiasSilva) [inline src/config/currencies.php:14]: are we sure on this values?

File: tests/TestCase.php [hunk 6]
```diff
@@ -40,9 +40,10 @@ public function loadConfiguration()
             $this->app->config = new Repository([
                 'countries' => $filesystem->getRequire(__DIR__ . '/../src/config/countries.php'),
                 'countries-partial' => $filesystem->getRequire(__DIR__ . '/../src/config/countries-partial.php'),
+                'currencies' => $filesystem->getRequire(__DIR__ . '/../src/config/currencies.php'),
             ]);
         } catch (FileNotFoundException $e) {
-            die("Package configuration files ['src/config/countries', 'src/config/countries-partial'] not found.");
+            die("Package configuration files not found: {$e->getMessage()}");
```
Comment (AndreiaMatiasSilva) [inline tests/TestCase.php:46]: 👍

File: src/CurrencyHelper.php (hunk 2 above)
Comment (francescopepe) [reply src/CurrencyHelper.php:24]: my bad

File: src/Facades/CurrencyHelper.php (hunk 4 above)
Comment (francescopepe) [reply src/Facades/CurrencyHelper.php:9]: Lines are too long and I personally don't like to put phpdoc for facade methods. But I can change them If you want :)

Comment (AndreiaMatiasSilva) [reply src/Facades/CurrencyHelper.php:9]: I see, that's ok we can keep it

File: src/config/currencies.php (hunk 5 above)
Comment (francescopepe) [reply src/config/currencies.php:14]: Your comment made me think that we don't need this config file actually. I'm pushing the new version

File: tests/Unit/CurrencyHelperTest.php [hunk 7]
```diff
@@ -170,6 +169,61 @@ public function testToFormatCanDisplayUpToSixDecimals()
...
+        $this->assertEquals($expected, $actual);
+
+        $nonBreakingSpace = "\xC2\xA0";
+        $expected = "kr{$nonBreakingSpace}542,67";
+        $actual = $currencyHelper->toStandardFormat(542.668, CurrencyCode::NORWEGIAN_KRONE, 'no');
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testToStandardFormatWithCustomPrecision()
```
Comment (AndreiaMatiasSilva) [inline tests/Unit/CurrencyHelperTest.php:193]: missing doc blocks

File: tests/Unit/CurrencyHelperTest.php [hunk 8]
```diff
@@ -170,6 +169,61 @@ public function testToFormatCanDisplayUpToSixDecimals()
         $this->assertEquals($expected, $actual);
     }
 
+    public function testToStandardFormat()
```
Comment (AndreiaMatiasSilva) [inline tests/Unit/CurrencyHelperTest.php:172]: missing doc blocks

File: src/CurrencyHelper.php [hunk 9]
```diff
@@ -39,8 +45,11 @@ public function toFormat($value, string $currencyCode = CurrencyCode::POUND_STER
...
      * @return string
      */
-    public function toFormatFromInt(int $value, string $currencyCode = CurrencyCode::POUND_STERLING, string $locale = 'en'): string
-    {
+    public function toFormatFromInt(
+        int $value,
+        string $currencyCode = CurrencyCode::POUND_STERLING,
+        string $locale = 'en'
+    ): string {
         return $this->toFormat($value / 100, $currencyCode, $locale);
```
Comment (AndreiaMatiasSilva) [inline src/CurrencyHelper.php:53]: here we can use `toStandardFormat` as `toFormat` will be deprecated

**Review by AndreiaMatiasSilva — APPROVED** [review]

**Pull request merged**